pkg/datastore/postgres/repositories/header_repository.go

```diff
@@ -107,3 +107,11 @@ func (repository HeaderRepository) MissingBlockNumbers(startingBlockNumber, endi
 	}
 	return numbers, nil
 }
+
+func (repository HeaderRepository) GetMostRecentHeaderBlockNumber() (int64, error) {
+	var blockNumber int64
+	err := repository.database.Get(&blockNumber,
+		`SELECT block_number FROM headers ORDER BY block_number DESC LIMIT 1`)
```

Comment:
Any idea how this compares performance-wise with `select max(block_number) from headers`?

Reply:
good question!  looks like `SELECT block_number FROM headers ORDER BY block_number DESC LIMIT 1` may be slightly more performant?

```
explain SELECT block_number FROM headers ORDER BY block_number DESC LIMIT 1;

Limit  (cost=0.43..0.47 rows=1 width=8)
  ->  Index Only Scan Backward using headers_block_number_eth_node_id_key on headers  (cost=0.43..113428.77 rows=2925850 width=8)

=======================

 explain select max(block_number) from headers;
Result  (cost=0.47..0.48 rows=1 width=8)
  InitPlan 1 (returns $0)
    ->  Limit  (cost=0.43..0.47 rows=1 width=8)
          ->  Index Only Scan Backward using headers_block_number_eth_node_id_key on headers  (cost=0.43..120743.40 rows=2925850 width=8)
                Index Cond: (block_number IS NOT NULL)
```